sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -1567,7 +1567,7 @@ class AdaptiveQueryExecSuite
         assert(!smj.head.isSkewJoin)
         // Both sides are coalesced.
         val customReaders = collect(smj.head) {
-          case c: CustomShuffleReaderExec if c.hasCoalescedPartition => c
+          case c: CustomShuffleReaderExec => c
```

Comment:
so this test actually do not coalesce the partitions?

Reply:
yes and no.  Sometimes empty partitions get dropped but not always.  The comment in the code just above https://github.com/apache/spark/pull/32776/files#r648846235 explains this exactly.

Reply:
I think here have two things:
1. hasCoalescedPartition; `CustomShuffleReaderExec` has some `CoalescedPartitionSpec`
2. coalesced; `CoalescedPartitionSpec` has some coalesced partition (exists `startReducerIndex + 1 != endReducerIndex`)

The `stringArgs` method we want to change is for option 2, so I think is it better to create new method `coalesced` to make them clearly ? cc @cloud-fan

Reply:
I'd like to keep one method. I don't think it's useful to only report if there is `CoalescedPartitionSpec`.

Reply:
one thing we can do is to update this test to make coalesce happen, like setting the `ADVISORY_PARTITION_SIZE_IN_BYTES` to a larger value.